lib/ex_unit/lib/ex_unit/formatter.ex

```diff
@@ -130,6 +130,37 @@ defmodule ExUnit.Formatter do
     format_assertion_error(%{}, struct, [], :infinity, fn _, msg -> msg end, "")
   end
 
+  def format_assertion_error(
+        test,
+        %{left: %ExUnit.Pattern{match_single?: false} = pattern, right: mailbox} = struct,
```

Comment:
What do you think about adding the concept of "mailbox" instead of `:match_single?`? I think `ExUnit.Pattern` should represent just the pattern, and not how it should be applied to the right side. The right side would be then either a value or a "mailbox".

Beside, this structure would make more sense if we finally end up reusing part of this logic in `FunctionClauseError`s:

> This approach can also be useful in other situations, for example, in the `UndefinedFunctionClauseError`, we could perform a better diff there too, as we show the patterns there.

In short words, I'm suggesting to rewrite this line to: 
```elixir
%{left: %ExUnit.Pattern{} = pattern, right: %ExUnit.Mailbox{} = mailbox} = struct
```
Thoughts?

Reply:
I like the suggestion.